Fix some jUnit tests on Linux

srstsavage opened the pull request:
Fix many path related and other issues with jUnit tests on Linux.

Note that these changes don't yet make every test pass, there are still issues with font differences in image generation, non-portable performance checks, and a handful of other differences.

srstsavage committed:
Fix some jUnit tests on Linux

Fix many path related and other issues with jUnit tests on Linux.

Note that these changes don't yet make every test pass, there
are still issues with font differences in image generation,
non-portable performance checks, and a handful of other differences.

srstsavage commented on src/test/java/gov/noaa/pfel/erddap/dataset/EDDTableFromNcFilesTests.java:
```diff
@@ -12390,13 +12390,13 @@ void testTablePseudoSourceNames() throws Throwable {
         +
         ",,degrees_east,degrees_north,UTC,m,degrees_north,,,,,degrees_true,\n" +
-        "41024,,-78.489,33.848,2014-01-15T00:00:00Z,7.0,33.848,NaN,33.848,down,,320,\"0.0, 27.0\"\n"
+        "41024,miniNdbc,-78.489,33.848,2014-01-15T00:00:00Z,7.0,33.848,NaN,33.848,down,,320,\"0.0, 27.0\"\n"
```
I'm curious if this change still works on Windows, or if it was working without it. My understanding is that `parentDir` should be getting populated here with `miniNdbc`, since that's what's parsed from the parent path in `test_migrations/src/test/java/testDataset/EDDTestDataset.java`:

```
<dataVariable>
  <sourceName>***pathName,.*/([a-zA-Z]*)/NDBC_.*_met\\.nc,1</sourceName>
  <destinationName>parentDir</destinationName>
  <dataType>String</dataType>
  <addAttributes>
    <att name="ioos_category">Identifier</att>
    <att name="long_name">Parent Directory</att>
  </addAttributes>
</dataVariable>
```

ChrisJohnNOAA replied:
My machine(s) do not have miniNdbc there. I'm looking into that and a few other test failures I'm getting with these changes.

ChrisJohnNOAA replied:
I made a pull request against your branch. Several of these broke tests on windows. I believe I've adjusted them so they work on both.

srstsavage replied:
All of the affected tests still seem to pass for me with your changes, so I merged. Thanks!

ChrisJohnNOAA committed:
Fix watch directory tests on windows

While linux may not send the modify events on deletion or the directory event, windows does

ChrisJohnNOAA committed:
Disable the display in browser, it causes tests to pause

ChrisJohnNOAA committed:
Have the path regex work for both windows and linux

ChrisJohnNOAA committed:
Fix the forwardSlashDir utility function

This is only used in one spot, so should be safe to modify.
Also it was previously checking if the string started with a / and if not, adding a slash to the end of it.

ChrisJohnNOAA committed:
Fix some of the EDDGridFromNcFilesTests on Linux

The wording of FileNotFoundException differs, so just make sure its that kind of exception. Mark testNcml flaky (for now), and fix the capitilzation of testGridNThreads for systems that are sensitive to that.

ChrisJohnNOAA committed:
Disable the performance part of persistentTableTests for now

ChrisJohnNOAA committed:
Change to using a temp dir for eddtablefromhttpgettests

srstsavage committed:
Merge pull request #1 from ChrisJohnNOAA/test_migrations_some_linux_fixes

Fixes so these work on Windows and Linux

srstsavage commented:
@ChrisJohnNOAA Your changes looked good to me and affected tests still passed on my Linux machine. Merged those!
